iso-authoryear.bbx

```diff
@@ -11,10 +11,12 @@
 % If there is no date field in the bib entry,
 % we should state there is no date by printing
 % a localisation string, e.g. [n.d.].
-\newcommand{\printextrayear}{%
-  \iffieldundef{year}
-    {\bibstring[\mkbibbrackets]{nodate}}%
-    {\printdateextra}%
+\newbibmacro*{date+extradate}{%
```

Comment:
I would prefer `\printlabeldate` here, just like in the standard styles. See https://github.com/moewew/biblatex-iso690/commit/a1fb7e1e4010990b4ce16fb667e72d1112abea91. But I don't know if I can get the `nodate` in brackets easily.

Reply:
`\printlabeldate` / `\printlabeldateextra` seems also not preserving the format of the `date` field defined by `\DeclareFieldFormat{date}` (brackets around circa date or space after dash in open range date).

Reply:
Yes, it obeys its own high-level formatting which can be set with `\DeclareFieldFormat{labeldate}`. But it should obey the general date-level formatting (which should include dashes for open ranges and similar stuff).

Reply:
I thought the solution would be to replace
* `\DeclareFieldFormat{date}` with `\DeclareFieldFormat{labeldate}` in `iso.bbx`,
* `\printdateextra` with `\printlabeldateextra` in `iso-authoryear.bbx`,
* `\printdate` with `\printlabeldate` in macro `\renewbibmacro*{date}` in `iso.bbx`,

for style `iso-authoryear` it seems to work, but not for `iso-numeric` -- the date is not printed at all. Is it another bug?

Why we couldn't replace only `\printdateextra` with `\printlabeldateextra` in moewew@a1fb7e1 (and possibly remove `\begingroup` and `\endgroup`)?

Reply:
Oh sorry, I only wanted the replacement in https://github.com/moewew/biblatex-iso690/commit/a1fb7e1. No other `\printdate...`s should be changed (the change you describe don't work in `iso-numeric`, because that style does not request `labeldate` info from Biber).

Going from
```latex
\newbibmacro*{date+extradate}{%
  \begingroup
    \iffieldundef{year}
      {\bibstring[\mkbibbrackets]{nodate}}
      {\printdateextra}%
  \endgroup
}
```
to
```latex
\newbibmacro*{date+extradate}{%
  \iffieldundef{labelyear}
    {\bibstring[\mkbibbrackets]{nodate}}
    {\printlabeldateextra}%
}
```
would be possible, but a bit pointless or redundant, since `\printlabeldate` already has code to handle `nodate`. We already use that in citations.

Reply:
Ok, but `labeldate` doesn't use the format of `date`, so I declared an alias for this field as not having a duplicated code for these fields. Already pushed 58788af.